backstory/src/demo/webapp/vision-demo.js

```diff
@@ -0,0 +1,163 @@
+        const imageListElement = document.getElementById('image-list');
+        imageListElement.innerHTML = '';
+
+        for (let i = 0; i < analyzedImagesObject.length; i++) {   
+          const imageUrl = analyzedImagesObject[i].imageUrl;
+          const labelsJsonArray = analyzedImagesObject[i].labelsJsonArray;
+          
+          // parse the labels json and pass it into the formatting array
+          const labelsObj = JSON.parse(labelsJsonArray);
+          const labels = labelsObj.labels;
+
+          imageListElement.appendChild(
+              createRow(imageUrl, labels));
+        }
+      });
+}
+
+/** 
+ * Formats the image url and labels array into HTML
+ *
+ * @param {string} imageUrl - the url of the analyzed image
+ * @param {object} labels - an array of labels & their other properties
+ * @return {object} - an Element containing the analyzed image with its labels and scores 
+ */
+function createRow(imageUrl, labels) {
+  const row = document.createElement('div');
+  row.classList.add('row');
+  row.classList.add('horizontal-flex-container');
+
+  const imageDiv = document.createElement('div');
+  imageDiv.classList.add('image-div');
+
+  // create a div to hold the image for this 
+  const image = document.createElement('img');
+  image.src = imageUrl;
+  image.classList.add('image');
+
+  // add image to its div, and add this div to container (row)
+  imageDiv.appendChild(image);
+  row.appendChild(imageDiv);
+
+  // create divs to hold labels
+  const labelsDiv = document.createElement('div');
+  labelsDiv.classList.add('labels-div');
+  const labelsTable = formatLabelsAsTable(labels);
+  labelsTable.id = 'labels-table';
+  
+  // add labels to row
+  labelsDiv.appendChild(labelsTable);
+  row.appendChild(labelsDiv);
+
+  return row;
+}
+
+/**
+ * Format labels as an HTML table.
+ * 
+ * @param {object} labels - an array of labels (as JSON)
+ * @return {object} - a table representing the labels array
+ */
+function formatLabelsAsTable(labels) {
+  const table = document.createElement('table');
+  table.classList.add('labels-table');
+
+  let index = 0;
+  
+  // add headers
+  const header = table.insertRow(index);
+  const labelHeader = document.createElement('th');
+  labelHeader.innerText = 'Label';
+  labelHeader.classList.add('labels-table-cell');
+  const scoreHeader = document.createElement('th');
+  scoreHeader.innerText = 'Score';
+  scoreHeader.classList.add('labels-table-cell');
+
+  header.appendChild(labelHeader);
+  header.appendChild(scoreHeader);
+
+  index++;
+
+  for (let i = 0; i < labels.length; i++) {
+    const label = labels[i];
+
+    const row = table.insertRow(index);
+
+    const labelData = row.insertCell(0);
+    labelData.innerText = label.description;
+    labelData.classList.add('labels-table-cell');
+
+    const scoreData = row.insertCell(1);
+    scoreData.innerText = `${(label.score * 100).toFixed(2)}%`;
+    scoreData.classList.add('labels-table-cell');
+
+    index++;
```

Comment:
Seems like if this was moved to the first line of this for loop, you could avoid the additional statement on line 124 above.

Reply:
That makes sense. I'll make the change.

Reply:
Done